contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -509,11 +509,30 @@ namespace eosiosystem {
       uint64_t  primary_key() const { return giftee.value; } // unique as one giftee can only hold gifted ram from one gifter
    };
 
+   struct [[eosio::table("peerkeysver"), eosio::contract("eosio.system")]] peer_keys_version {
+      uint64_t   version;                                  // version incremented every time peer_keys_table is modified
+
+      uint64_t primary_key() const { return 0; }           // table has just one row
```

Comment:
Usually for a single row table the pattern is to use a singleton. For example, the globals,
https://github.com/eosnetworkfoundation/eos-system-contracts/blob/c66e447b7883d457557398907be5ad72f808fe86/contracts/eosio.system/include/eosio.system/eosio.system.hpp#L454-L460

Reply:
I was a little worried about `singleton` since you wrote that:
```
find()s are one of the most expensive host functions a contract can call, and cdt's singleton impl is woof in this regard. eosio system contract: 12 find()s on every single action to deal with singletons
```

But if you say `singleton` is better I'm happy to change.

Reply:
singleton really seems like the right pattern here

The performance woof can be resolved in cdt fairly easily some day. Actually, we can probably resolve the problem in nodeos too by short circuiting redundant  lookups in the iterator cache.